### PR TITLE
Documentation for development with k8s

### PR DESCRIPTION
**Pull request by snwagh**
## Description
* Adding `07-deploy-devspace.ipynb` which closes this [issue](https://github.com/OpenMined/Heartbeat/issues/1217)
* Few additions to `03-deploy-k8s-k3d.ipynb`:
  * Resource constraint deployment instructions
  * Custom password setup during deployment or extracting randomized password post deployment

## Testing
* Tested over bash until Docker account got frozen. Works.

### CODE REVIEW

**Comment by khoaguin** [comment]
Hi @snwagh can we add the newly made commands to launch different types of nodes into the tutorial, for example:

1. For Launching a Domain
```zsh
CLUSTER_NAME=testdomain1 CLUSTER_HTTP_PORT=9082 tox -e dev.k8s.launch.domain
```
2. For Launching a Gateway
```zsh
CLUSTER_NAME=testgateway1 CLUSTER_HTTP_PORT=9081 tox -e dev.k8s.launch.gateway
```
3. For Launching a Enclave
```zsh
CLUSTER_NAME=testenclave1 CLUSTER_HTTP_PORT=9083 tox -e dev.k8s.launch.enclave
```
4. For launching nodes with `hotreload`:
```zsh
tox -e dev.k8s.domain -- hotreload
```

**Comment by snwagh** [comment]
@khoaguin Great suggestion, added these.

File: notebooks/tutorials/deploy/07-deploy-devspace.ipynb [hunk 1]
```diff
@@ -4,8 +4,132 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "TODO"
+    "# Developing over PySyft\n",
+    "This guide is meant for developers trying to deploy PySyft locally on a Kubernetes cluster. PySyft is a powerful framework for privacy-preserving machine learning, and deploying it on Kubernetes allows an easy way to quickly try out the full PySyft stack on your own system. This guide will walk you through the process step by step. \n",
```
Comment (IrinaMBejan) [inline notebooks/tutorials/deploy/07-deploy-devspace.ipynb]: I think privacy-preserving ML is only a dimension of what one could do with PySyft - to keep the messaging uniform, I'd either extend it as `privacy-preserving data analysis and machine learning` or `conducting privacy-preserving research on non-public data`

File: notebooks/tutorials/deploy/07-deploy-devspace.ipynb [hunk 2]
```diff
@@ -4,8 +4,132 @@
...
+     "languageId": "plaintext"
+    }
+   },
+   "source": [
+    "In order to perform an isolated setup, create a virtual environment. Below are some instructions to get started (example `environment.yml` and `requirements.txt` file are [provided below](#dependency-files)):\n",
+    "| Conda | Venv |\n",
+    "|:---|:---|\n",
+    "| 1. Create  `environment.yml`  with name  `syft-dev`  <br /> 2. Run  `conda env create -f environment.yml`  <br /> 3. Run  `conda activate syft-dev` | 1. Create a  `requirements.txt`  file <br /> 2. Run  `python -m venv syft-dev`  <br /> 3. Run  `.\\syft-dev\\Scripts\\activate`  for Windows or <br />  `source syft-dev/bin/activate`  for Mac/Linux <br /> 4.  `pip install -r requirements.txt`  <br /> 5. `pip install -e \"packages/syft[ dev ]\"` |\n",
+    "\n",
+    "If you would like to a non-isolated install then install the dependencies using [brew](https://brew.sh/) as follows:   \n",
```
Comment (IrinaMBejan) [inline notebooks/tutorials/deploy/07-deploy-devspace.ipynb]: to do a*

**Review by IrinaMBejan — CHANGES_REQUESTED** [review]
This is excellent! Just a few nit comments.

**Review by itstauq — APPROVED** [review]

**Review by IrinaMBejan — APPROVED** [review]

**Pull request merged**